### ISSUE TITLE
`flake8-copyright` does not yet support `flake8>=6`

### ISSUE THREAD

**Issue by ca-scribner** [canonical/template-operator#39]
### Bug Description

The release of `flake8` 6 is not yet supported by flake8-copyright.  A PR with a fix is available [here](https://github.com/savoirfairelinux/flake8-copyright/pull/20/commits/c0b1b7e466d9baeb85b4ec9af57126463392048a), but the repo has been a few years since a release so not sure if they'll be quick to update.  

fwiw, I have tried the fix by pinning to the specific commit (pinned to git+https://github.com/savoirfairelinux/flake8-copyright.git@c0b1b7e466d9baeb85b4ec9af57126463392048a), and it worked in my testing.

### To Reproduce

```
git clone git@github.com:canonical/template-operator
cd template-operator
tox -e lint
```

### Environment

All

### Relevant log output

```shell
Fails with messages like:

  File "/home/scribs/code/canonical/kubeflow-tensorboards-operator/update-to-1.6.1/template-operator/.tox/lint/lib/python3.9/site-packages/flake8_copyright.py", line 30, in register_opt
    parser.add_option(*args, **kwargs)
  File "/home/scribs/code/canonical/kubeflow-tensorboards-operator/update-to-1.6.1/template-operator/.tox/lint/lib/python3.9/site-packages/flake8/options/manager.py", line 281, in add_option
    self._current_group.add_argument(*option_args, **option_kwargs)
  File "/home/scribs/anaconda3/lib/python3.9/argparse.py", line 1423, in add_argument
    raise ValueError('%r is not callable' % (type_func,))
ValueError: 'int' is not callable
```
where the root cause is flake8 not using optparse.
```


### Additional context

_No response_

**Comment by jnsgruk** [comment]
https://github.com/canonical/charmcraft/pull/932 

I've been moving to ruff, given recent instability with flake8 and plugins :)

**Comment by jnsgruk** [comment]
Fixed in #45
